tests/common/test_transaction.py

```diff
@@ -481,6 +475,15 @@ def test_asset_link_deserialization_with_empty_payload():
     assert asset_link == expected
 
 
+def test_asset_link_from_inputs(utx, transfer_utx):
+    from bigchaindb.common.transaction import AssetLink
+
+    expected = AssetLink(utx.id)
+    asset_link = AssetLink.from_inputs(utx, transfer_utx)
```

Comment:
Fair enough, I guess this is actually doing more validation than building anything 😉 . I'll take a look into the driver to see how it uses `Transaction.transfer()` and try to remove this.